app/javascript/packs/session-timeout-ping.ts

```diff
@@ -100,36 +94,24 @@ function success(data: PingResponse) {
 }
 
 function ping() {
-  const request = new XMLHttpRequest();
-  request.open('GET', '/active', true);
-
-  request.onload = function () {
-    try {
-      success(JSON.parse(request.responseText));
-    } catch (error) {
-      notifyNewRelic(request, error, 'ping');
-    }
-  };
+  try {
+    request('/active').then(success);
+  } catch (error) {
+    notifyNewRelic(error, 'ping');
+  }
```

Comment:
how come this this is a `try { ... } catch { .. }` instead of a `.catch()` on the promise?

Reply:
Partly trying to minimize the diff (unsuccessfully). Partly because I initially reimplemented this as async/await before realizing that would change the behavior with the setTimeout below. I can switch to the chain catch.

Reply:
I don't feel strongly, I was just worried that if the promise threw an error, it would be in the other thread and not caught here? Especially since there's no `await`? But my mental model of async JS is lacking

Reply:
Updated in c9594b15d

Reply:
For what it's worth, I have my eye to refactor this entire script as a custom element. There's a lot of issues here with excessive polling. I described most of this at https://github.com/18F/identity-idp/pull/7966#issuecomment-1478458412, but another issue I discovered is that the `keepalive` logic will [cause](https://github.com/18F/identity-idp/blob/c9594b15d369cbe505748e3789bc5649a7caf6ce/app/javascript/packs/session-timeout-ping.ts#L107) a duplicate polling interval to start in parallel to the main "ping" polling. I'm hopeful that refactoring it as a class instance could allow for some improvements in maintaining this as a shared timeout.